### ISSUE TITLE
Make generated jar builds reproducible

### ISSUE THREAD

**Issue by nemphys** [beryx/badass-jlink-plugin#113]
While trying to integrate Update4j in my project, I was banging my head why it would decide that an update is available after consecutive builds with no changes to the code.

Digging a little deeper, I discovered that update4j checks the filename, the file size and the *md5* hash of each project file. After a little testing, I realised that some of my project's jars had a different md5 hash after each build, even though nothing had changed in their content.

This led me to enable the relevant settings in gradle (see https://dzone.com/articles/reproducible-builds-in-java) and now the issue is fixed for all jars except for the ones created by the badass jlink plugin.

Since this ruins my whole auto-update feature, is there any chance it can be fixed?

EDIT: Just in case it helps, I found this: https://github.com/gradle/gradle/blob/master/subprojects/core/src/main/java/org/gradle/api/internal/file/archive/ZipCopyAction.java (how gradle handles the timestamp issue)

**Comment by nemphys** [comment]
And I also found this: https://github.com/Johni0702/gradle-reproducible-builds-plugin.
Testing now, if it works, it seems quite easy to implement in your plugin.

**Comment by siordache** [comment]
Implemented in 2.17.2.
The jars created by the badass-jlink plugin are now always reproducible. For the other tasks you still need to configure:
```
tasks.withType(AbstractArchiveTask) {
    preserveFileTimestamps = false
    reproducibleFileOrder = true
}
```

**Comment by nemphys** [comment]
As always, you rock :-)
